lib/modules/manager/gomod/extract.ts

```diff
@@ -46,7 +50,7 @@ function getGoDep(lineNumber: number, goVer: string): PackageDependency {
     depType: 'golang',
     currentValue: goVer,
```

Comment:
There won't be any way to differentiate between the go.mod directive and the go toolchain directive. I think we need to differentiate by depType by making this new one different